src/csharp/Microsoft.Spark/ML/Feature/Base.cs

```diff
@@ -109,7 +116,32 @@ private static T WrapAsType(JvmObjectReference reference)
                         (parameters[0].ParameterType == typeof(JvmObjectReference));
                 });
 
-            return (T)constructor.Invoke(new object[] {reference});
+            return (T)constructor.Invoke(new object[] { reference });
+        }
+    }
+
+    /// <summary>
+    /// DotnetUtils is used to hold basic general helper functions that
+    /// are used within ML scope.
+    /// </summary>
+    public class DotnetUtils
```

Comment:
Are you planning to use this anywhere else besides the 1 reference in Pipeline.GetStages?  If not then I suggest moving the `GetUnderlyingType` method to Pipeline.cs.  Also I suggest making it private unless you are expecting users to consume this API.

Reply:
Yes this is needed inside of some other ML classes in order to fetch the real type from the jvm side. I'll make it private though.

Reply:
Actually I'll make it internal so that within the dll other classes can access it